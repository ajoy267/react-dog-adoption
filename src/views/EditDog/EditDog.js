import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import DogForm from '../../components/DogForm/DogForm';
import { deleteDog, fetchDogById, updateDog } from '../../services/dog';

export default function EditDog() {
  const [id, setId] = useState('');
  const [name, setName] = useState('');
  const [image, setImage] = useState('');
  const [age, setAge] = useState('');
  const [breed, setBreed] = useState('');
  const [bio, setBio] = useState('');

  const params = useParams();
  useEffect(() => {
    const fetchData = async () => {
      const resp = await fetchDogById(params.id);
      setId(resp.id);
      setName(resp.name);
      setImage(resp.image);
      setAge(resp.age);
      setBreed(resp.breed);
      setBio(resp.bio);
    };
    fetchData();
  }, [params.id]);

  const handleSubmit = async (e) => {
    e.preventDefault();
    await updateDog(params.id, id, name, image, age, breed, bio);
  };

  const handleDelete = async (e) => {
    e.preventDefault();
    await deleteDog(params.id);
  };
  return (
    <div>
      <h1>Edit Dog</h1>
      <DogForm
        id={id}
        setId={setId}
        name={name}
        setName={setName}
        image={image}
        setImage={setImage}
        age={age}
        setAge={setAge}
        breed={breed}
        setBreed={setBreed}
        bio={bio}
        setBio={setBio}
        handleSubmit={handleSubmit}
        handleDelete={handleDelete}
      />
    </div>
  );
}
