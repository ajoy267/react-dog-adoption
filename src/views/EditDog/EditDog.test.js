import { MemoryRouter, Route } from 'react-router-dom';
import EditDog from '../EditDog/EditDog';
import { render, screen } from '@testing-library/react';

test('should render an individual dog', async () => {
  // render the component
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/206/edit']}>
      <Route path="/dogs/:id/edit" component={EditDog} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('Charles');
  expect(container).toMatchSnapshot();
});
