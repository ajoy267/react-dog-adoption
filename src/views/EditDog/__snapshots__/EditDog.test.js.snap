// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an individual dog 1`] = `
<div>
  <div>
    <h1>
      Edit Dog
    </h1>
    <div
      class="dog-form"
    >
      <form>
        <h2>
          Dog Details
        </h2>
        <div
          class="form-control"
        >
          <label>
            Id:
          </label>
          <input
            placeholder="Id"
            type="numebr"
            value="206"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Name:
          </label>
          <input
            placeholder="Name"
            type="text"
            value="Charles"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Image:
          </label>
          <input
            placeholder="Image URL"
            type="text"
            value="https://placedog.net/800/640?id=206"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Age:
          </label>
          <input
            placeholder="Age"
            type="number"
            value="7"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Breed:
          </label>
          <input
            placeholder="Breed"
            type="text"
            value="Labrador"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Bio:
          </label>
          <textarea
            placeholder="Bio"
          >
            Charles isn't your typical elderly dog, He enjoys chasing random strangers such as the mailman and bringing you sticks.
          </textarea>
        </div>
        <div>
          <button>
            Save
          </button>
        </div>
        <div>
          <button>
            Delete
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
